### ISSUE TITLE
Headless support - Clarification

### ISSUE THREAD

**Issue by amin212121** [Adyen/adyen-shopware6#246]
**Describe the bug**
Does support Cards payment for PWA? Because now I get error when try use it for PWA

**Comment by Suzell** [comment]
Hey, this should be working yes. What is the error you see and how can we reproduce it please?

**Comment by Morerice** [comment]
Hi @amin212121,

Could you please confirm if this is still an issue on your end? We currently have multiple merchants that are using a headless integration with our Shopware6 plugin that have not reported this issue. However if you still have this issue, could you provide some further details?

Thanks,
Jean
Adyen

**Comment by amin212121** [comment]
Hey @Morerice thank you. I just asking because didn't find information in documentation about headless and in April I didn't know how to use it but now I understand all. Thx

**Comment by amin212121** [comment]
> Hey, this should be working yes. What is the error you see and how can we reproduce it please?

all ok. thx you

**Comment by amin212121** [comment]
You can close the issue

**Comment by amin212121** [comment]
@Suzell and @Morerice hello again.

I want to use RatePay method via Adyen Shopware plugin could you please provide me documentation with rest API how I can do it?
small info I use PWA but not VUE.js I use React

**Comment by bitfactory-michel-lubbers** [comment]
Hi, 
I'm using shopware-pwa with adyen.
And I'm trying to get all the payment methods to work.
However when i use ideal I'm not getting the correct view.
It only shows a button with the following messages 'Preautorisatie uitvoeren met undefined'
It is not much saying error.

- No banks are shown in the view.
![image](https://user-images.githubusercontent.com/100353261/212275696-b4289b6c-dfae-42f7-a2e9-423cde435758.png)

Do you have some extra info that i need to look at to get those payments to work correctly?

**Comment by peterojo** [comment]
Hi @amin212121 and @bitfactory-michel-lubbers!
Here's some [documentation](https://docs.adyen.com/plugins/shopware-6/headless-integration) for using headless storefront with our plugin. 
TL;DR
Our endpoints in the plugin can provide you with all the extra configuration data for rendering the payment component on your frontend but you do have to render the payment method fields on your frontend. Our recommendation is to use our [Web Components](https://docs.adyen.com/online-payments/web-components) for this.
Please read the docs for more info.

**Comment by GM-Alex** [comment]
@peterojo It seems broken for payments which redirects the user, like Klarna. Normally the AsynchronousPaymentHandlerInterface::pay method should return the redirect URL to the PSP for example Klarna. I made a pull request for this: https://github.com/Adyen/adyen-shopware6/pull/361

**Comment by adyen-integrations-support** [comment]
Hello @GM-Alex, @peterojo, @bitfactory-michel-lubbers,

Please note that some time has passed since the last update on this issue. Is this issue still relevant, or can it be closed?

Thank you,

Kind regards.
